native_extension/GoogleSignInANE/src/com/tuarua/google/GoogleSignIn.as

```diff
@@ -67,28 +113,33 @@ public final class GoogleSignIn extends EventDispatcher {
      * ability to disconnect their Google account from the app. If the user deletes their account, you must
      * delete the information that your app obtained from the Google APIs. */
     public function revokeAccess():void {
-        GoogleSignInANEContext.validate();
-        var ret:* = GoogleSignInANEContext.context.call("revokeAccess");
+        _context.validate();
+        var ret:* = _context.context.call("revokeAccess");
         if (ret is ANEError) throw ret as ANEError;
     }
 
     /** */
     public function handle(url:String, sourceApplication:String):void {
-        GoogleSignInANEContext.validate();
-        var ret:* = GoogleSignInANEContext.context.call("handle", url, sourceApplication);
+        _context.validate();
+        var ret:* = _context.context.call("handle", url, sourceApplication);
         if (ret is ANEError) throw ret as ANEError;
     }
 
-    /** Disposes the ANE */
+    /** Disposes default ANE instance */
     public static function dispose():void {
-        if (GoogleSignInANEContext.context) {
-            GoogleSignInANEContext.dispose();
+        if (_shared) {
+            _shared.dispose();
         }
     }
 
-    /** set the API key if not using Firebase resources */
+    public function dispose():void {
+        _context.dispose();
+    }
+
+    /** set the serverClientId if not using Firebase resources */
     public static function set apiKey(value:String):void {
-        _apiKey = value;
+        _serverClientId = value;
```

Comment:
why is serverClientId being set in a setter for apiKey?

Reply:
That was named wrongly as apiKey at first. That value was used as `clientId` of a web server in the native library. I decided to keep naming for compatibility.